mlflow/llama_index/tracer.py

```diff
@@ -293,3 +340,99 @@ def _extract_token_usage(
                     if (v := additional_kwargs.get(k)) is not None:
                         usage[k] = v
             return usage
+
+
+_StreamEndEvent = Union[LLMChatEndEvent, LLMCompletionEndEvent, ExceptionEvent]
+
+
+class StreamResolver:
+    """
+    A class is responsible for closing the pending streaming spans that are waiting
+    for the stream to be exhausted. Once the associated stream is exhausted, this
+    class will resolve the span, as well as recursively resolve the parent spans
+    that returns the same (or derived) stream.
+    """
+
+    def __init__(self, client: mlflow.MlflowClient):
+        self._client = client
+        self._span_id_to_span_and_gen: Dict[str, Tuple[LiveSpan, Generator]] = {}
+
+    def is_streaming_result(self, result: Any) -> bool:
+        return (
+            inspect.isgenerator(result)  # noqa: SIM101
+            or isinstance(result, (StreamingResponse, AsyncStreamingResponse))
+            or isinstance(result, StreamingAgentChatResponse)
+            or (isinstance(result, TaskStepOutput) and self.is_streaming_result(result.output))
+        )
+
+    def register_stream_span(self, span: LiveSpan, result: Any) -> bool:
+        """
+        Register the pending streaming span with the associated generator.
+
+        Args:
+            span: The span that has a streaming output.
+            result: The streaming result that is being processed.
+
+        Returns:
+            True if the span is registered successfully, False otherwise.
+        """
+        if inspect.isgenerator(result):
+            stream = result
+        elif isinstance(result, (StreamingResponse, AsyncStreamingResponse)):
+            stream = result.response_gen
+        elif isinstance(result, StreamingAgentChatResponse):
+            stream = result.chat_stream
+        elif isinstance(result, TaskStepOutput):
+            stream = result.output.chat_stream
+        else:
+            _logger.warning(
+                f"Unsupported streaming response type: {type(result)}. The span will "
```

Comment:
This is a bit of an internal implementation warning that a dev would have context for, but it's confusing to a user. Could we change this to be user-friendly? 
"Unable to record traces for <x>. {type(result)} is not supported for MLflow tracing." 
The 'ending immediately' warning might make users confused as they'll still see the actual streamed tokens output.

Reply:
Makes sense. The message is a bit tricky as the trace will actually be recorded (with serialized generator as output), but I will try to make it more user friendly.

Reply:
Should we add the message to the trace as an event (similar to logging a raised stack trace message in the OTeL spec)?

Reply:
That sounds like a good idea:)

Reply:
On second thought, this line is also like "unexpected reach" thing like the one below (cuz I check `is_streming_result` before entering the function). Lemme replace with exception for internal surface:)